athena-federation-integ-test/src/main/java/com/amazonaws/athena/connector/integ/ConnectorStackAttributesProvider.java

```diff
@@ -0,0 +1,109 @@
+    private final String id;
+    private final String lambdaFunctionName;
+    private final Map<String, Object> testConfig;
+    private final Optional<PolicyDocument> connectorAccessPolicy;
+    private final Map<String, String> environmentVariables;
+    private final ConnectorPackagingAttributes connectorPackagingAttributes;
+    private final Optional<ConnectorVpcAttributes> connectorVpcAttributes;
+
+    protected ConnectorStackAttributesProvider(final Construct scope, final String id, final String lambdaFunctionName,
+                                               final Map<String, Object> testConfig,
+                                               final Optional<PolicyDocument> connectorAccessPolicy,
+                                               final Map<String, String> environmentVariables)
+    {
+        this.scope = scope;
+        this.id = id;
+        this.lambdaFunctionName = lambdaFunctionName;
+        this.testConfig = testConfig;
+        this.connectorAccessPolicy = connectorAccessPolicy;
+        this.environmentVariables = environmentVariables;
+        this.connectorPackagingAttributes = ConnectorPackagingAttributesProvider.getAttributes();
+        this.connectorVpcAttributes = ConnectorVpcAttributesProvider.getAttributes(testConfig);
+
+        setUpEnvironmentVars();
+    }
+
+    /**
+     * Sets defaults for environment variables (spill_bucket, spill_prefix, disable_spill_encryption) if not provided
+     * by connector.
+     * @return A Map containing the environment variables key-value pairs.
+     * @throws RuntimeException The spill_bucket is neither specified in the environment vars nor the test config file.
+     */
+    private void setUpEnvironmentVars()
+            throws RuntimeException
+    {
+        // Check for missing spill_bucket
+        if (!environmentVariables.containsKey(LAMBDA_SPILL_BUCKET_TAG)) {
+            // Add missing spill_bucket environment variable from test config file
+            Object spillBucket = testConfig.get(LAMBDA_SPILL_BUCKET_TAG);
+            if (!(spillBucket instanceof String) || ((String) spillBucket).isEmpty()) {
+                throw new RuntimeException(
+                        "spill_bucket must be specified in environment var or test-config.json.");
+            }
+            environmentVariables.put(LAMBDA_SPILL_BUCKET_TAG, (String) spillBucket);
+        }
+
+        // Check for missing spill_prefix
+        if (!environmentVariables.containsKey(LAMBDA_SPILL_PREFIX_TAG)) {
+            // Add missing spill_prefix environment variable
+            environmentVariables.put(LAMBDA_SPILL_PREFIX_TAG, "athena-spill");
+        }
+
+        // Check for missing disable_spill_encryption environment variable
+        if (!environmentVariables.containsKey(LAMBDA_DISABLE_SPILL_ENCRYPTION_TAG)) {
+            // Add missing disable_spill_encryption environment variable
+            environmentVariables.put(LAMBDA_DISABLE_SPILL_ENCRYPTION_TAG, "false");
+        }
```

Comment:
would it be better to throw here than have defaults? we provide defaults in SAR only for convenience in the UI... i'm not sure thats appropriate here since we have a robust config file infra you've added.

Reply:
moved to config file.